docs/tutorial/typesfuns.rst

```diff
@@ -1182,3 +1183,30 @@ matching ``let`` and lambda bindings. It will *only* work if:
 - The type of the result is "known". i.e. the type of the expression
   can be determined *without* type checking the ``case``-expression
   itself.
+
+Totality
+========
+
+A *total* function is one that is defined for all inputs of the correct
+type: that is, it is guaranteed not to crash. For example, a function
```

Comment:
How would you feel about a less formal definition to accompany the diverging discussion? Maybe something like [Edwin's definition here](https://edwinb.wordpress.com/2013/02/19/on-partial-functions-in-idris/)?


> * A function which terminates for possible inputs, or
> * A function which is guaranteed to produce some output before making a recursive call